compiler/rustc_hir_analysis/src/check/intrinsicck.rs

```diff
@@ -69,6 +69,7 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
             // `!` is allowed for input but not for output (issue #87802)
             ty::Never if is_input => return None,
             ty::Error(_) => return None,
+            _ if ty.references_error() => return None,
```

Comment:
The latter arm implies the former arm

```suggestion
            _ if ty.references_error() => return None,
```